kategory/src/main/kotlin/kategory/instances/Function1Instances.kt

```diff
@@ -0,0 +1,31 @@
+package kategory
+
+interface Function1Instances<P> :
+        Functor<Function1.F>,
+        Applicative<Function1.F>,
+        Monad<Function1.F>,
+        MonadReader<Function1.F, P> {
+
+    override fun ask(): HK<Function1.F, P> =
+            { a: P -> a }.k()
+
+    override fun <A> local(f: (P) -> P, fa: HK<Function1.F, A>): Function1<P, A> =
+            f.andThen { fa.ev().invoke(it) }.k()
+
+    override fun <A> pure(a: A): Function1<P, A> =
+            { _: P -> a }.k()
+
+    override fun <A, B> map(fa: HK<Function1.F, A>, f: (A) -> B): HK<Function1.F, B> =
```

Comment:
Same as above, the parameter info isn't available to the compiler. It infers Function1<B, B>.